### ISSUE TITLE
Change timeout code to use threads

### ISSUE THREAD

**Issue by jairideout** [caporaso-lab-graveyard/clout#1]
The code that handles timeouts (i.e. in setting up/tearing down the cluster, or running the test suites) currently uses Python's signal.alarm. This is problematic because if more than one automated testing system is running at the same time on the same system, the timeout code only works for one of the scripts.

The solution is to modify run_test_suites.py to use threads, similar to the technique described here:

http://stackoverflow.com/a/4825933

Basically, spawn a new thread that executes the necessary commands that have the potential of hanging indefinitely, and then wait for the thread to complete for a specified amount of time. This approach should play well with multiple copies of run_test_suites.py running simultaneously.

This solution also has the advantage of being more platform-independent than using the signal module and SIGALRM (it should work on Windows).

**Comment by jairideout** [comment]
Pull request was merged, so closing this issue.
